File: backend/app/schemas/idea.py
import uuid
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel


class ComponentBase(BaseModel):
    name: str
    description: str = ""
    priority: str = "should_have"
    estimated_cost: Decimal | None = None
    deadline: datetime | None = None


class ComponentCreate(ComponentBase):
    pass


class ComponentUpdate(BaseModel):
    name: str | None = None
    description: str | None = None
    priority: str | None = None
    status: str | None = None
    estimated_cost: Decimal | None = None
    deadline: datetime | None = None


class ComponentResponse(ComponentBase):
    id: uuid.UUID
    idea_id: uuid.UUID
    status: str
    shareability_score: float
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class IdeaCreate(BaseModel):
    title: str
    description: str = ""
    deadline: datetime | None = None
    total_budget: Decimal | None = None
    budget_currency: str = "USD"


class IdeaUpdate(BaseModel):
    title: str | None = None
    description: str | None = None
    status: str | None = None
    deadline: datetime | None = None
    total_budget: Decimal | None = None
    budget_currency: str | None = None


class IdeaResponse(BaseModel):
    id: uuid.UUID
    tenant_id: uuid.UUID
    creator_id: uuid.UUID | None
    title: str
    description: str
    status: str
    deadline: datetime | None
    total_budget: Decimal | None
    budget_currency: str
    version: int
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class IdeaDetailResponse(IdeaResponse):
    components: list[ComponentResponse] = []

File: frontend/src/types/idea.ts
export interface Idea {
  id: string;
  tenant_id: string;
  creator_id: string;
  title: string;
  description: string;
  status: "draft" | "active" | "archived";
  deadline: string | null;
  total_budget: number | null;
  budget_currency: string;
  version: number;
  created_at: string;
  updated_at: string;
}

export interface IdeaComponent {
  id: string;
  idea_id: string;
  name: string;
  description: string;
  estimated_cost: number | null;
  priority: "must_have" | "should_have" | "nice_to_have";
  status: "proposed" | "approved" | "in_progress" | "done";
  deadline: string | null;
  shareability_score: number;
  created_at: string;
}

export interface Budget {
  id: string;
  idea_id: string;
  total_amount: number;
  currency: string;
  line_items: BudgetLineItem[];
}

export interface BudgetLineItem {
  id: string;
  budget_id: string;
  description: string;
  amount: number;
  component_id: string | null;
}
